.github/workflows/daily-tests.yml

```diff
@@ -0,0 +1,68 @@
+name: daily-tests
+
+permissions:
+  contents: read
+
+on:
+  schedule:
+    # We run these tests on a daily basis (at a time slightly offset from the
+    # top of the hour), as their runtime is either too long for the usual per-PR
+    # CI, or because they rely on external 3rd party services that can be flaky.
+    - cron: '15 18 * * *'
+
+jobs:
+  build:
+    name: Daily Tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # test a bunch of toolchains on ubuntu
+        rust:
+          - stable
+          - beta
+          - nightly
+        os: [ubuntu-20.04]
+        # but only stable on macos/windows (slower platforms)
+        include:
+          - os: macos-latest
+            rust: stable
+          - os: windows-latest
+            rust: stable
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Install ${{ matrix.rust }} toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.rust }}
+
+      - name: Install cargo hack
+        uses: taiki-e/install-action@cargo-hack
+
+      - name: Build main crate
+        run: cargo build
+
+      - name: Run connect tests
+        run: cargo test --manifest-path=connect-tests/Cargo.toml
+        env:
+          RUST_BACKTRACE: 1
+
+      - name: Check simple client
+        run: cargo run --bin simpleclient
+
+      - name: Check limited client
+        run: cargo run --bin limitedclient
+
+      - name: Check simple 0rtt client
+        run: cargo run --bin simple_0rtt_client
+
+      # Test the server_acceptor binary builds - we invoke with --help since it
+      # will run a server process that doesn't exit when invoked with no args
+      - name: Check server acceptor
+        run: cargo run --bin server_acceptor -- --help
+
+      - name: Check feature powerset
+        run: cargo hack check --feature-powerset --no-dev-deps
```

Comment:
It might make sense to pull this (and maybe the non connect test task) into their own `job`, but for now I've done it all in one for simplicity's sake.

Reply:
I think this does make sense, mainly to avoid running the feature-powerset thing on multiple platforms. Just ubuntu-latest should be enough?

Reply:
I split the workflow up into three jobs: one for the connect tests, one for the example binary smoke tests, and one for the feature powerset. The feature powerset job only runs on ubuntu with stable rust.